lib/wpc_proto/internal_modules/proto_otap.c

```diff
@@ -0,0 +1,115 @@
+/* Wirepas Oy licensed under Apache License, Version 2.0
+ *
+ * See file LICENSE for full license details.
+ *
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "wpc.h"
+#include "proto_otap.h"
+#include "proto_config.h"
+#include <pb_encode.h>
+#include <pb_decode.h>
+#include "platform.h"
+#include "common.h"
+
+#define LOG_MODULE_NAME "otap_proto"
+#define MAX_LOG_LEVEL INFO_LOG_LEVEL
+#include "logger.h"
+
+static app_scratchpad_status_t otap_status = {0};
+
+static bool initialize_otap_variables()
+{
+    if (WPC_get_local_scratchpad_status(&otap_status) != APP_RES_OK)
+    {
+        LOGE("Cannot get local scratchpad status\n");
+        return false;
+    }
+    return true;
+}
+
+bool Proto_otap_init(void)
+{
+    /* Read initial config from sink */
+    if (!initialize_otap_variables())
+    {
+        LOGE("All the otap settings cannot be read\n");
+    }
+    return true;
+}
+
+void Proto_otap_close(void)
+{
+    WPC_unregister_for_data();
+}
+
+wp_ScratchpadType otap_convert_type(uint8_t scrat_type)
+{
+    switch (scrat_type)
+    {
+        case 1:
+            return wp_ScratchpadType_PRESENT;
+        case 2:
+            return wp_ScratchpadType_PROCESS;
+        case 0:
+        default:
+            return wp_ScratchpadType_BLANK;
+    }
+}
+
+wp_ScratchpadStatus otap_convert_status(uint8_t scrat_status)
+{
+    switch (scrat_status)
+    {
+        case 0:
+            return wp_ScratchpadStatus_SUCCESS;
+        case 255:
+            return wp_ScratchpadStatus_NEW;
+        default:
+            return wp_ScratchpadStatus_ERROR;
+    }
+}
+
+app_proto_res_e Proto_otap_handle_get_scratchpad_status(wp_GetScratchpadStatusReq *req,
+                                                        wp_GetScratchpadStatusResp *resp)
+{
+    app_res_e res = APP_RES_OK;
+
+    // TODO: Add some sanity checks
+
+    // Do we need to refresh otap status ?
```

Comment:
I don't think it is required, but should be read at every stack status changed. As status will change mainly with a reboot of the node

Reply:
Yes, it's linked with the previous comment with initial read. I've removed it in the last commit, when moved in config.